assignment-1/Functions.java

```diff
@@ -0,0 +1,271 @@
+                    int playerRuns = Integer.parseInt(individualField[4]);
+                    Double playerAverage = Double.parseDouble(individualField[5]);
+                    Double playerStrikeRate = Double.parseDouble(individualField[6]);
+                    int playerWickets = Integer.parseInt(individualField[7]);
+                    allPlayers.add(new Players(playerName, playerTeam, playerRole, playerMatches, playerRuns, playerAverage, playerStrikeRate, playerWickets));
+                }
+            }
+
+            //allTeams object
+
+            for (Players player : allPlayers) {
+                boolean flag = false;
+                String temp = player.getTeam();
+                for (Team t : allTeams) {
+                    if (t.Name.equals(temp)) {
+                        flag = true;
+                        t.players.add(player);
+                        t.Name = temp;
+                        break;
+                    }
+                }
+                if (!flag) {
+                    ArrayList<Players> play = new ArrayList<Players>();
+                    play.add(new Players(player.getName(), player.getTeam(), player.getRole(), player.getMatches(), player.getRuns(), player.getAverage(), player.getStrikeRate(), player.getWickets()));
+                    allTeams.add(new Team(temp, play));
+                }
+            }
+
+            br.close();
+        } catch (FileNotFoundException fnfe) {
+            System.out.println(fnfe.getMessage());
+        } catch (IOException ioe) {
+            System.out.println(ioe.getMessage());
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    //Second Task
+    public static void matchFixture() throws FileNotFoundException {
+        try {
+            PrintWriter pw = new PrintWriter(new File("fixture.csv"));
+            StringBuilder sb = new StringBuilder();
+
+            sb.append("Sr.No");
+            sb.append(",");
+            sb.append("Guest Team");
+            sb.append(",");
+            sb.append(("Home Team"));
+            sb.append(",");
+            sb.append("Venue-Ground");
+            sb.append("\n");
+            Set<String> hash_Set = new HashSet<String>();
+            for (Players t : allPlayers) {
+                hash_Set.add(t.getTeam());
+            }
+            Integer srNo = 0;
+            for (String s : hash_Set) {
+                for (String t : hash_Set) {
+                    if (s != t) {
+                        srNo++;
+//                        System.out.println(s + " " + t);
+                        sb.append(srNo);
+                        sb.append(",");
+                        sb.append(s);
+                        sb.append(",");
+                        sb.append(t);
+                        sb.append(",");
+                        sb.append("null");
+                        sb.append("\n");
+                    }
+                }
+            }
+
```

Comment:
This whole method should have been segregated into 3 methods performing 3 operations.
1. creation of the fixtures
2. string them in a list
3. writing them to a file